lib/puppetx/puppetlabs/strings/yard/templates/default/puppetnamespace/setup.rb

```diff
@@ -1,7 +1,87 @@
 include T('default/module')
 
+require File.join(File.dirname(__FILE__),'../html_helper')
+require File.join(File.dirname(__FILE__),'../template_helper')
+
 def init
   sections :header, :box_info, :pre_docstring, T('docstring'),
     :method_summary, [:item_summary],
     :method_details_list, [T('method_details')]
+
+  @methods = object.children
+  @template_helper = TemplateHelper.new
+end
+
+def header
+  if @methods[0]['puppet_4x_function']
+    @header_text = "Puppet 4 Functions"
+  else
+    @header_text = "Puppet 3 Functions"
+  end
```

Comment:
So how I _think_ this works is that this template will be used twice, once to generate dos for all the puppet 3 functions (which will all be displayed on one webpage) and once to generate all the docs for the puppet 4 functions (which will all be displayed on another webpage). When I poked around in the debugger it seemed that this template was used twice, and each time the array contained only 3x or 4x functions.

This is 100% just from observation though so I could be wrong